### DISCUSSION TITLE
Rails public/vite folder contains source files after bin/vite build

### DISCUSSION THREAD

**Discussion by fvsch** [ElMassimo/vite_ruby#293] [Q&A] [answered]
Hi there!

I hesitated creating an issue because I’m not sure if that's expected behavior.

In a Rails app with Vite Ruby, running `bin/vite build` adds a bunch of files to `public/vite`, including:

1. `manifest.json` and an empty `manifest-assets.json`
2. an `assets` folder with compiled CSS/JS/images and source maps
3. and a bunch of source files including ERB and TypeScript files

<img alt="" width="532" src="https://user-images.githubusercontent.com/243601/199476742-0c5a6629-026a-4afb-bb69-0a290f7d947f.png">

That third category seems both unnecessary and a risk of data leak?

Those source files are also not part of the files listed in the terminal (by Vite or Rollup, not sure) upon a successful build, so I’m wondering if Vite Ruby is copying them over (and why!).

Setting `build.sourcemap` to `false` in `vite.config.ts` does not change anything for these files (but it does prevent emitting `.map` files).

Am I missing something?

**Comment by ElMassimo** [comment, marked as answer]
Hi Florens!

Haven't seen this before, so it's likely related to the app's configuration or dir structure.

If I had to guess based on the dir names, it looks like the [`sourceCodeDir`](https://vite-ruby.netlify.app/config/index.html#sourcecodedir) is somehow ending up in Vite's output.

Vite will copy anything that is inside its [`public` dir](https://vitejs.dev/config/shared-options.html#publicdir), which is one of the possible causes.

Vite Ruby will also process any [`additionalEntrypoints`](https://vite-ruby.netlify.app/config/index.html#additionalentrypoints). You might want to check that you don't have an `assets` dir inside [`sourceCodeDir`](https://vite-ruby.netlify.app/config/index.html#sourcecodedir) that contains code instead of images/stylesheets, or configure it to exclude that dir.

I'd recommend checking both your Vite Ruby and Vite config, and try using the default configuration (if you have customized it).

**Reply by fvsch** [reply]
Hmm I think the issue is that we have a structure like:

```
app/
  assets/
    …stuff for sprockets…
  frontends/
    entrypoints/…
    some-frontend/…
    public/…
config/
  vite.json
public/
vite.config.ts
```

And we configured the `sourceCodeDir` to be `app/frontends`, because setting it to e.g. `app` would make Vite Ruby or Vite (not sure) try to compile a bunch of SCSS code from `app/frontends/stylesheets`.

Apparently you should not have any folder named "assets" OR "public" at the root of your `sourceCodeDir`. ^^

The simple solution might be to rename `app/frontends/public` to something else like `app/frontends/publicpages`. I tried to avoid that because of the Git and organisational noise of moving a lot of source files, but it would fix the issue.

**Reply by ElMassimo** [reply]
Makes sense. You can configure [`publicDir`](https://vitejs.dev/config/shared-options.html#publicdir) in Vite to be something else, like `static`:

```ts
// vite.config.ts
export default defineConfig({
  publicDir: 'static',
})
```

That way you don't need to move files.

**Reply by fvsch** [reply]
Thanks!

Turns out you can set it to `publicDir: false` to disable the feature completely, and that works pretty well for us.
